core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -951,15 +951,22 @@ class KafkaApis(val requestChannel: RequestChannel,
               new OffsetFetchResponse(Errors.NONE, authorizedPartitionData.asJava, header.apiVersion)
             }
           } else {
-            val (authorizedPartitions, unauthorizedPartitions) = offsetFetchRequest.partitions.asScala
-              .partition(authorizeTopicDescribe)
-            val (error, authorizedPartitionData) = coordinator.handleFetchOffsets(offsetFetchRequest.groupId,
-              Some(authorizedPartitions))
+            val (authorizedPartitions, unauthorizedPartitions) =
+              offsetFetchRequest.partitions.asScala.partition(authorizeTopicDescribe)
+            val (error, authorizedPartitionData) =
+              coordinator.handleFetchOffsets(offsetFetchRequest.groupId, Some(authorizedPartitions))
             if (error != Errors.NONE)
               offsetFetchRequest.getErrorResponse(error)
             else {
+              val authorizedAndValidatedPartitionData = authorizedPartitionData.map {
+                case (topicPartition, offsetInfo) =>
+                  if (metadataCache.getPartitionInfo(topicPartition.topic, topicPartition.partition).isEmpty)
```

Comment:
Thanks for the explanation. I think this change make sense, particular because we want Kafka to be able to respond with `authorizedAndValidatedPartitionData` when user is not authorized to describe the topic. There is security reason why we don't explicitly tell user `not authorized to describe`.

If we were to make the behavior consistent by returning `OffsetFetchResponse.UNKNOWN_PARTITION` when the partition doesn't exist, should we check the existence of the partition even before calling and checking the error from `coordinator.handleFetchOffsets(...)`?

Also I have a minor concern. I am not sure if the name `authorizedAndValidatedPartitionData` is very intuitive. Developer may be wondering what is `authorizedAndUnValidatedPartitionData`.